apps/studio/src/features/editing-experience/components/form-builder/renderers/controls/JsonFormsLinkArrayControl.tsx

```diff
@@ -0,0 +1,369 @@
+    return (
+      <EditLinkItem
+        renderers={renderers}
+        cells={cells}
+        visible={visible}
+        schema={schema}
+        uischema={getChildUiSchema(composePaths(path, `${selectedIndex}`))}
+        path={composePaths(path, `${selectedIndex}`)}
+        handleRemoveItem={handleRemoveItem(path, selectedIndex)}
+        onBack={() => setSelectedIndex(undefined)}
+      />
+    )
+  }
+
+  return (
+    <VStack spacing="0.375rem" align="start">
+      <VStack justify="space-between" align="start" w="full">
+        <HStack w="full" justifyContent="space-between">
+          <VStack align="start" spacing="0.25rem">
+            <Text textStyle="subhead-1">{label}</Text>
+
+            {description && (
+              <Text textStyle="body-2" textColor="base.content.default">
+                {description}
+              </Text>
+            )}
+
+            {arraySchema.maxItems && (
+              <Text textStyle="body-2" textColor="base.content.medium">
+                {data}/{arraySchema.maxItems} links added
+              </Text>
+            )}
+          </VStack>
+
+          <Tooltip
+            label={
+              arraySchema.maxItems && data >= arraySchema.maxItems
+                ? `You can only place up to ${arraySchema.maxItems} links.`
+                : undefined
+            }
+            hasArrow
+          >
+            <Button
+              variant="clear"
+              size="xs"
+              leftIcon={<Icon as={BiPlusCircle} />}
+              onClick={addItem(path, createDefaultValue(schema, rootSchema))}
+              isDisabled={
+                arraySchema.maxItems ? data >= arraySchema.maxItems : false
+              }
+            >
+              Add a link
+            </Button>
+          </Tooltip>
+        </HStack>
+      </VStack>
+      <DragDropContext onDragEnd={onDragEnd}>
+        <Droppable droppableId="blocks">
+          {({ droppableProps, innerRef, placeholder }) => (
+            <VStack
+              {...droppableProps}
+              align="baseline"
+              w="100%"
+              h="100%"
+              spacing={0}
+              ref={innerRef}
+              mt="-0.25rem"
+            >
+              {data === 0 && (
+                <Flex
+                  alignItems="center"
+                  flexDir="column"
+                  px="1.5rem"
+                  p="3.75rem"
+                  mt="0.25rem"
+                  justifyContent="center"
+                  w="100%"
+                >
+                  <Text
+                    textStyle="subhead-1"
+                    textColor="base.content.default"
+                    textAlign="center"
+                  >
+                    Items you add will appear here
+                  </Text>
+                </Flex>
+              )}
+
+              {[...Array(data).keys()].map((index) => {
+                const childPath = composePaths(path, `${index}`)
```

Comment:
issue: i think there's a double render issue with errors, see attached video. i think it's because the index changes and the key is dependent on the index.

https://github.com/user-attachments/assets/1a9d796f-a641-426a-b610-1ecdd350f2fa